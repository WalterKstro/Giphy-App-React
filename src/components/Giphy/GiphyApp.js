import './GiphyApp.css'
import {useState} from 'react'
import AddCategory from '../Add/AddCategory'
import GridCategory from '../Category/CategoryGrid'


const GiphyApp = ({categoriesDefault = []}) => {
    const [categories, setCategories] = useState( categoriesDefault )
    return (
        <>
            <AddCategory 
                setCategories={ setCategories }
            />
            
            <div className="category">
                {
                    categories.map( category => <GridCategory category={category} key={category} />)
                }
            </div>
        </>
    )
}


export default GiphyApp