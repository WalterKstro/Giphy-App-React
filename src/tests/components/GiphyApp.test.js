import { shallow } from "enzyme"
import GiphyApp from "../../components/Giphy/GiphyApp"

describe( 'Testing component root GiphyApp', () => {

    test('should show a snaptshot', () => {
        const component = shallow(<GiphyApp/>)

        expect( component).toMatchSnapshot()
    })
    
    test('should show the same number of components <CategoryGrid/> as the length of the elements of prop array', () => {
        const categories = ['One Punch Man','Naruto','Futurama']
        const component = shallow(<GiphyApp categoriesDefault={categories} />)
        expect(component).toMatchSnapshot()
        expect(component.find('GridCategory').length).toBe(categories.length)
    })
    

} )