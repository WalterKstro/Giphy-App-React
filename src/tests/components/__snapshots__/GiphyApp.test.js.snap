// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component root GiphyApp should show a snaptshot 1`] = `
<Fragment>
  <AddCategory
    setCategories={[Function]}
  />
  <div
    className="category"
  />
</Fragment>
`;

exports[`Testing component root GiphyApp should show the same number of components <CategoryGrid/> as the length of the elements of prop array 1`] = `
<Fragment>
  <AddCategory
    setCategories={[Function]}
  />
  <div
    className="category"
  >
    <GridCategory
      category="One Punch Man"
      key="One Punch Man"
    />
    <GridCategory
      category="Naruto"
      key="Naruto"
    />
    <GridCategory
      category="Futurama"
      key="Futurama"
    />
  </div>
</Fragment>
`;
